packages/cli/src/config/loadWorldConfig.ts

```diff
@@ -0,0 +1,179 @@
+import { z, ZodError } from "zod";
+import { fromZodErrorCustom, UnrecognizedSystemErrorFactory } from "../utils/errors.js";
+import { BaseRoute, ObjectName } from "./commonSchemas.js";
+import { loadConfig } from "./loadConfig.js";
+import { validateEthereumAddressOrSystemName } from "./validation.js";
+
+// Object names have to start with a capital letter and only contain letters, numbers and underscores
+const SystemName = ObjectName;
+
+// Base routes are ordinary routes with an empty string as a valid value
+const SystemRoute = BaseRoute.optional();
+
+// Each element in the SystemAccessList must be a valid Ethereum address or a valid system name
+const SystemAccessList = z
+  .array(z.string())
+  .default([])
+  .superRefine((arr, ctx) => {
+    arr.forEach((element) => validateEthereumAddressOrSystemName(element, ctx));
+  });
+
+// The system config is a combination of a route config and access config
+const SystemConfig = z.intersection(
+  z.object({
+    route: SystemRoute,
+  }),
+  z.discriminatedUnion("openAccess", [
+    z.object({
+      openAccess: z.literal(true),
+    }),
+    z.object({
+      openAccess: z.literal(false),
+      accessList: SystemAccessList,
+    }),
+  ])
+);
+
+// The parsed world config is the result of parsing the user config
+export const WorldConfig = z.object({
+  baseRoute: BaseRoute.default(""),
+  worldContractName: z.string().default("World"),
+  overrideSystems: z.record(SystemName, SystemConfig).default({}),
```

Comment:
What about just `systems`? 

and `exclude` could be an option, e.g.
```
systems: {
  SomeSystemToOverride: {...},
  SomeSystemToExclude: { exclude: true }
}
```

Reply:
imo `overrideSystems` makes it more obvious that systems not mentioned here are included by default. If it was just called `systems`, I would assume only systems mentioned here to be deployed. In `resovleWorldConfig` we then use the `overrideSystems` and default systems to build the full `systems` key, so the resulting config object only includes the `systems` key (not `overrideSystems` or `excludeSystems`)